src/dotnet/commands/dotnet-install/InstallCommand.cs

```diff
@@ -0,0 +1,137 @@
+            ParseResult result = parser.ParseFrom("dotnet install", args);
+
+            result.ShowHelpOrErrorIfAppropriate();
+
+            AppliedOption parseResult = result["dotnet"]["install"]["tool"];
+
+            var packageId = parseResult.Arguments.Single();
+            var packageVersion = parseResult.ValueOrDefault<string>("version");
+
+            FilePath? configFile;
+
+            var configFilePath = parseResult.ValueOrDefault<string>("configfile");
+            configFile = configFilePath != null
+                ? new FilePath(configFilePath)
+                : GetConfigFileProbingCurrentDirectory();
+
+            var framework = parseResult.ValueOrDefault<string>("framework");
+
+            var executablePackagePath = new DirectoryPath(new CliFolderPathCalculator().ExecutablePackagesPath);
+
+            var toolConfigurationAndExecutableDirectory = ObtainPackage(
+                packageId,
+                packageVersion,
+                configFile,
+                framework,
+                executablePackagePath);
+
+            DirectoryPath executable = toolConfigurationAndExecutableDirectory
+                .ExecutableDirectory
+                .WithSubDirectories(
+                    toolConfigurationAndExecutableDirectory
+                        .Configuration
+                        .ToolAssemblyEntryPoint);
+
+            var shellShimMaker = new ShellShimMaker.ShellShimMaker(executablePackagePath.Value);
+            var commandName = toolConfigurationAndExecutableDirectory.Configuration.CommandName;
+            shellShimMaker.EnsureCommandNameUniqueness(commandName);
+
+            shellShimMaker.CreateShim(
+                executable.Value,
+                commandName);
+
+            EnvironmentPathFactory
+                .CreateEnvironmentPathInstruction()
+                .PrintAddPathInstructionIfPathDoesNotExist();
+
+            Reporter.Output.WriteLine(
+                $"{Environment.NewLine}The installation succeeded. If there is no other instruction. You can type the following command in shell directly to invoke: {commandName}");
+
+            return 0;
+        }
+
+        private static FilePath? GetConfigFileProbingCurrentDirectory()
+        {
+            // https://github.com/NuGet/Home/issues/6199
+            // Pending NuGet change
+
+            var settings = RestoreSettingsUtils.ReadSettings(
+                solutionDirectory: Directory
+                    .GetCurrentDirectory(), // current directory instead of fake project as start
+                restoreDirectory: Directory.GetCurrentDirectory(), // current directory instead of fake project as start
+                restoreConfigFile: null, // no explicit config file that why we prob it
+                machineWideSettings: new Lazy<IMachineWideSettings>(() => new XPlatMachineWideSetting()));
+
+            var nugetconfigfilePath = SettingsUtility.GetConfigFilePaths(settings).FirstOrDefault(); // it is ordered
```

Comment:
What is the reason for getting only the first nuget.config file? Typically the entire hierarchy is used through ISettings

Reply:
The problem is I cannot pass ISettings down. The interface for restore is the temp project. I need to find a way to serialize ISettings into the temp project or passing it via MSBuild props. However, restore target takes only one config file as parameter, so I will pick the first one(it will be the config file in current directory, or something without explicit configfile path passed in anyway) or null (equal to no explicit --configfile).

Reply:
How about setting `<RestoreSolutionDirectory>` in the project? 

Passing a single config as if it were the only config used is likely to cause problems. Often times users have nuget.org defined in the their user wide config and other settings in the working directory config file. If you were to select just the working directory one it would miss sources.